File: 05.02.二进制数转字符串/05.02.go
func printBin(num float64) string {
	res := "0."
	for num > 0 {
		num = num * 2
		if num >= 1 {
			res = res + "1"
			num = num - 1
		} else {
			res = res + "0"
		}

		if len(res) > 32 {
			return "ERROR"
		}
	}
	return res
}

File: 05.02.二进制数转字符串/05.02.cpp
class Solution {
public:
    string printBin(double num) {
        string res = "0.";
        while (num != 0) {
            num *= 2;
            if (num >= 1) {
                res += "1";
                num -= 1;
            } else {
                res += "0";
            }

            if (res.size() > 32) {
                return "ERROR";
            } 
        }
        return res;
    }
};